rust/datafusion/src/execution/physical_plan/planner.rs

```diff
@@ -153,33 +202,28 @@ impl PhysicalPlanner for DefaultPhysicalPlanner {
                     .collect::<Result<Vec<_>>>()?;
 
                 let initial_aggr = HashAggregateExec::try_new(
+                    AggregateMode::Partial,
```

Comment:
An improvement for a subsequent PR would be to only skip the partial aggregate if the input distribution is ` Distribution::SinglePartition`

Reply:
Yes, that's a great point.